Core/StitchCoreSDK/Sources/StitchCoreSDK/Internal/Net/AppMetadata.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+
+/// Application Metadata
+public final class AppMetadata: Codable, CustomStringConvertible {
+    let deploymentModel: String
+    let location: String
+    let hostname: String
+    
+    public var description: String {
+        let encoder = JSONEncoder()
+        
+        guard let data = try? encoder.encode(self),
+              let str = String(data: data, encoding: .utf8) else {
+            return "<json serialization error>"
```

Comment:
[design] This should really be an impossible point to reach, but this block implies it is possible. For about the same amount of lines, you could just use a string or multiline string:

```
public lazy var description = """
        AppMetadata={
            deploymentModel=\(deploymentModel),
            location=\(location)
            hostname=\(hostname)
    }
"""
```

Since this class is virtually immutable, this value can also be lazily computed, since we don't need to recompute description after the first time.

As a bonus, here's a fancy but non-performant way to do descriptions as well:
```
public lazy var description: String = Mirror.init(
            reflecting: self
        ).children.reduce("AppMetadata={") { (res, child) -> String in
            "\(res)\(child.label!)=\(child.value), "
        }.dropLast(2) + "}"
```

Reply:
I see your point, but idk if I agree. The first non-reflection approach has to change if we ever add more fields to app metadata, and in the case we do add fields that can't be serialized, we would want the description to fail with a JSON serialization error rather than omit then entirely because we forgot to update the `description` getter

Reply:
so while the case is technically impossible to reach for now, it could become possible with future changes that might not touch that particular method.

Reply:
There's a few reasons why my first example is the most efficient and most idiomatic approach. Ranked by importance:
- `CustomStringConvertible` is the canonical representation of some instance. Think about this protocol more in the context of an `asString` method as opposed to a `toString` method. An `ObjectId` would be canonised as its hex string value, not the extended JSON value. Because as a String, that's what it actually is. A `Point` class would be `(\(x), \(y))`.
- This struct is intended to be immutable. It should be lazily computed to avoid unnecessary computation.
- The reason we are encountering all of these weird `try` issues around getters/setters is because we're abusing the concept of computed properties. They are still supposed to be simple properties– doing computations that can `throw` breaches the intended use of computed properties, especially if you need to resurface the error. Acceptable use cases would be a default value if a `try` fails.
- Serialising to JSON every time you want to get the string description is unnecessarily cumbersome.

If we add fields that cannot be serialised (which should never be possible, we are the one that chooses what can or cannot be serialised), failing in the `description` is absolutely the wrong place to be failing. If we add fields that can be serialised and are added to the description, forgetting to update the `description` getter is an unacceptable reason to break all of the rules I've listed above. The same level of developer responsibility would apply to Java's `toString`, which generally does not use JSON or reflection.

Reply:
fair enough, all this makes sense

Reply:
removed for now